### ISSUE TITLE
Support for default value inside properties expression

### ISSUE THREAD

**Issue by angelozerr** [eclipse/lsp4mp#201]
Parser should support the default value for expression to fix https://github.com/redhat-developer/vscode-quarkus/issues/392

I think xe should ignore unkwown error for property expression which define default value.

**Comment by jamesfalkner** [comment]
works for me!

**Comment by angelozerr** [comment]
>  I would just try some particularly bad values (eg. ${:DBUSER::foo} etc.)

Good catch! There is no expections but we should report an error, because in this case we have:

 * reference property which is empty
 * default value which is `DBUSER::foo`

It's forbidden to have an empty reference property. We should manage a diagnostic for that. I will work on it.

**Comment by gian1200** [comment]
Is this supposed to handle only Strings? It fails with `quarkus.log.level`

![image](https://user-images.githubusercontent.com/405347/159553939-a0bdd61c-4922-4361-b91d-c37cdf2a0452.png)

This appears with Quick Fix:
![image](https://user-images.githubusercontent.com/405347/159554173-1ca44642-8654-46c5-a780-2cf0699ef173.png)

**Comment by angelozerr** [comment]
@gian1200 which version of vscode quarkus are you using? This feature has not been released

**Comment by gian1200** [comment]
The last public one (AFAIK)

Version: 1.9.0

https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus

![image](https://user-images.githubusercontent.com/405347/159572576-27bf24e7-ec63-4812-9e98-34f2b7c1d935.png)

**Comment by angelozerr** [comment]
@[gian1200](https://github.com/gian1200) I tried with 1.10.0 which has been released and there is the same problem.

Could you create an issue please.

**Comment by gian1200** [comment]
@angelozerr done! https://github.com/eclipse/lsp4mp/issues/225

**Comment by angelozerr** [comment]
Thanks ! I have started something. Lets continue the discussion in your issueq
